qutebrowser/keyinput/keyutils.py

```diff
@@ -72,6 +81,27 @@ def is_modifier_key(key):
     return key in _MODIFIER_MAP
 
 
+def _remap_unicode(key, text):
+    """Work around QtKeyEvent's bad values for high codepoints.
+
+    QKeyEvent handles higher unicode codepoints poorly (see
+    QTBUG-72776). It uses UTF-16 to handle key events, and for
+    higher codepoints that require UTF-16 surrogates (e.g. emoji
+    and some CJK characters), it sets the keycode to just the
+    upper half of the surrogate, which renders it useless, and
+    breaks UTF-8 encoding, causing crashes. So we detect this
+    case, and reassign the key code to be the full Unicode
+    codepoint, which we can recover from the text() property,
+    wihch has the full character."""
+    if _is_surrogate(key):
+        if len(text) != 1:
```

Comment:
I'm not too familiar with any of this logic, but if this function gets an empty string it will throw an error saying "key had too many characters". Would it be possible to define that case a little better (unless we should never get empty strings in here due to other reasons)?

Reply:
According to a message above, this is an unexpected case.